core/src/services/onedrive/builder.rs

```diff
@@ -20,25 +20,47 @@ use std::fmt::Debug;
 use std::fmt::Formatter;
 
 use log::debug;
+use serde::Deserialize;
 
 use super::backend::OnedriveBackend;
 use crate::raw::normalize_root;
+use crate::raw::ConfigDeserializer;
 use crate::raw::HttpClient;
 use crate::Scheme;
 use crate::*;
 
+/// Config for [OneDrive](https://onedrive.com) backend support.
+#[derive(Default, Deserialize)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct OnedriveConfig {
+    /// bearer access token for OneDrive
+    pub access_token: Option<String>,
+    /// root path of OneDrive folder.
+    pub root: Option<String>,
+}
+
+impl Debug for OnedriveConfig {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("OnedriveConfig")
+            .field("root", &self.root)
+            .finish_non_exhaustive()
+    }
+}
+
 /// [OneDrive](https://onedrive.com) backend support.
 #[doc = include_str!("docs.md")]
 #[derive(Default)]
 pub struct OnedriveBuilder {
-    access_token: Option<String>,
-    root: Option<String>,
+    config: OnedriveConfig,
     http_client: Option<HttpClient>,
 }
 
 impl Debug for OnedriveBuilder {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("Backend").field("root", &self.root).finish()
+        f.debug_struct("Backend")
+            .field("root", &self.config.root)
```

Comment:
`root` is included in `OnedriveConfig`. You just need `.field("config", &self.config)` here

Reply:
but access_token is in config too

Reply:
> `root` is included in `OnedriveConfig`. You just need `.field("config", &self.config)` here

you are right